tests/test.cpp

```diff
@@ -150,8 +150,9 @@ flatbuffers::unique_ptr_t CreateFlatBufferTest(std::string &buffer) {
 void AccessFlatBufferTest(const uint8_t *flatbuf, size_t length) {
 
   // First, verify the buffers integrity (optional)
-  flatbuffers::Verifier verifier(flatbuf, length);
+  flatbuffers::Verifier verifier(flatbuf, length + 123);
```

Comment:
Adding a random value here is not great. Maybe run the verifier a second time for the purpose of computing the size, and comment what you're doing.

See, this here shows that this functionality is problematic. If we don't know `length`, we have to pass an arbitrary number that is the biggest we think the buffer can be, to allow it to compute the real size. But in doing so, it may touch unallocated memory beyond the buffer. So rather than an arbitrary number, you MUST pass the size here of the buffer it sits in.

So a more accurate test would be to put the given FlatBuffer into a `std::vector` twice, append additional random data (e.g. a string), and then pass the size of that vector to the verifier, and see that it comes back with the length of the original string.